src/TCommandLine.cpp

```diff
@@ -279,192 +212,227 @@ bool TCommandLine::event( QEvent * event )
+            ke->accept();
+            return true;
+            break;
+
+        case Qt::Key_C:
+            if (ke->modifiers() & Qt::ControlModifier) {
+                if (mpConsole->console->mSelectedRegion != QRegion(0, 0, 0, 0)) {
+                    mpConsole->console->copySelectionToClipboard();
                     ke->accept();
                     return true;
                 }
+            }
+            break;
+
+        default:
+
+            if (mpKeyUnit->processDataStream(ke->key(), (int)ke->modifiers())) {
+                ke->accept();
+                return true;
+            } else {
+                QPlainTextEdit::event(event);
+                adjustHeight();
+
+                if (mpHost->mAutoClearCommandLineAfterSend)
+                    mHistoryBuffer = -1;
                 else
+                    mHistoryBuffer = 0;
+                if (mTabCompletionOld != toPlainText()) //text() )
                 {
-                    QPlainTextEdit::event( event );
-                    adjustHeight();
-
-                    if( mpHost->mAutoClearCommandLineAfterSend )
-                        mHistoryBuffer = -1;
-                    else
-                        mHistoryBuffer = 0;
-                    if( mTabCompletionOld != toPlainText() )//text() )
-                    {
-                        mUserKeptOnTyping = true;
-                        mAutoCompletionCount = -1;
-                    }
-                    else
-                    {
-                        mUserKeptOnTyping = false;
-                    }
-                    spellCheck();
-                    return false;
+                    mUserKeptOnTyping = true;
+                    mAutoCompletionCount = -1;
+                } else {
+                    mUserKeptOnTyping = false;
                 }
+                spellCheck();
+                return false;
+            }
         }
-
     }
 
-    return QPlainTextEdit::event( event );
+    return QPlainTextEdit::event(event);
 }
 
-void TCommandLine::focusInEvent( QFocusEvent * event )
+void TCommandLine::focusInEvent(QFocusEvent* event)
 {
     textCursor().movePosition(QTextCursor::Start);
-    textCursor().movePosition(QTextCursor::Right, QTextCursor::KeepAnchor, mSelectedText.length() );
+    textCursor().movePosition(QTextCursor::Right, QTextCursor::KeepAnchor, mSelectedText.length());
 
     mpConsole->console->forceUpdate();
     mpConsole->console2->forceUpdate();
-    QPlainTextEdit::focusInEvent( event );
+    QPlainTextEdit::focusInEvent(event);
 }
 
-void TCommandLine::focusOutEvent( QFocusEvent * event )
+void TCommandLine::focusOutEvent(QFocusEvent* event)
 {
-    if( textCursor().hasSelection() )
-    {
+    if (textCursor().hasSelection()) {
         mSelectionStart = textCursor().selectionStart();
         mSelectedText = textCursor().selectedText();
-    }
-    else
-    {
+    } else {
         mSelectionStart = 0;
         mSelectedText = "";
```

Comment:
If this is a `QString` `::clear()` avoids raw string literals... mind you so does `std::string::clear()` but those aren't going to show up when we start playing _hunt the UI QStrings_!